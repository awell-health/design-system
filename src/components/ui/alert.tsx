import * as React from "react";
import { cva, type VariantProps } from "class-variance-authority";

import { cn } from "@/lib/utils";
import { Icon } from "./icon";

const alertVariants = cva("alert p-4 rounded-lg border gap-3", {
  variants: {
    variant: {
      default: "bg-slate-50 border-slate-300 text-slate-600 [&h3]:text-slate-700",
      primary: "bg-blue-50 border-blue-300 text-blue-600 [&h3]:text-blue-700",
      warning: "bg-yellow-50 border-yellow-300 text-yellow-600 [&h3]:text-yellow-700",
      error: "bg-red-50 border-red-300 text-red-600 [&h3]:text-red-700",
    },
  },
  defaultVariants: {
    variant: "default",
  },
});

interface Props
  extends React.HTMLAttributes<HTMLDivElement>,
    VariantProps<typeof alertVariants> {
  title: string;
  showIcon?: boolean
}

const Alert = React.forwardRef<HTMLDivElement, Props>(
  ({ className, variant, title, children, showIcon = false, ...props }, ref) => {
    const icon = variant === 'error' || variant === 'warning' ? 'RiAlertLine' : 'RiInformationLine'
    const iconClassNames = cn({
      'fill-slate-500': variant === 'default',
      'fill-blue-500': variant === 'primary',
      'fill-yellow-500': variant === 'warning',
      'fill-red-500': variant === 'error'
    })

    return (
      <div
        role="alert"
        ref={ref}
        className={cn(alertVariants({ variant }), className)}
        {...props}
      >
        <div className="flex self-start pt-[5px]">
          {showIcon && <Icon icon={icon} size={16} className={iconClassNames} />}
        </div>
        <div>
          <h3 className="font-medium">{title}</h3>
          {children && <div className="font-normal leading-tight text-xs">{children}</div>}
        </div>
      </div>
    )
  }
);

export { Alert };
