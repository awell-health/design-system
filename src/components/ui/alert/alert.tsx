import * as React from 'react';
import { cva, type VariantProps } from 'class-variance-authority';

import { cn } from '@/lib/utils';
import { Icon } from '../icon';

const alertVariants = cva('alert p-4 rounded-lg border gap-3 text-sm', {
  variants: {
    variant: {
      default: 'bg-slate-50 border-slate-300 text-slate-600 [&h3]:text-slate-700',
      primary: 'bg-blue-50 border-blue-300 text-blue-700 [&h3]:text-blue-800',
      warning: 'bg-yellow-50 border-yellow-300 text-yellow-600 [&h3]:text-yellow-700',
      error: 'bg-red-50 border-red-300 text-red-600 [&h3]:text-red-700'
    }
  },
  defaultVariants: {
    variant: 'default'
  }
});

interface Props extends React.HTMLAttributes<HTMLDivElement>, VariantProps<typeof alertVariants> {
  title: string;
  showIcon?: boolean;
  children?: React.ReactNode;
  titleClassName?: string;
}

const Alert = React.forwardRef<HTMLDivElement, Props>(
  // eslint-disable-next-line
  ({ className, variant, title, children, showIcon = false, titleClassName, ...props }, ref) => {
    const icon = variant === 'error' || variant === 'warning' ? 'RiAlertLine' : 'RiInformationLine';
    const iconClassNames = cn({
      'fill-slate-500': variant === 'default',
      'fill-blue-500': variant === 'primary',
      'fill-yellow-500': variant === 'warning',
      'fill-red-500': variant === 'error'
    });

    return (
      <div role='alert' ref={ref} className={cn(alertVariants({ variant }), className)} {...props}>
        <div className='flex self-start pt-0.5'>
          {showIcon && <Icon icon={icon} size={16} className={iconClassNames} />}
        </div>
        <div className='flex flex-col gap-1'>
          <h3 className={cn('font-medium', titleClassName)}>{title}</h3>
          {children && <div className='font-normal'>{children}</div>}
        </div>
      </div>
    );
  }
);

Alert.displayName = 'Alert';

export { Alert };
