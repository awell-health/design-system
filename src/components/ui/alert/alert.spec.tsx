import { render } from '@testing-library/react';
import { expect, it, describe } from 'vitest';

import { Alert } from './alert';

describe('Alert', () => {
  it('match snapshot', () => {
    const result = render(<Alert title='Alert Title' showIcon={true} />);

    expect(result).toMatchSnapshot();
  });
});
