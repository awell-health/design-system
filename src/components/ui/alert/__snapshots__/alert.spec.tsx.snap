// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Alert > match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="alert p-4 rounded-lg border gap-3 text-sm bg-slate-50 border-slate-300 text-slate-600 [&h3]:text-slate-700"
        role="alert"
      >
        <div
          class="flex self-start pt-0.5"
        >
          <svg
            class="remixicon "
            fill="default"
            height="16"
            viewBox="0 0 24 24"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 22C6.47715 22 2 17.5228 2 12C2 6.47715 6.47715 2 12 2C17.5228 2 22 6.47715 22 12C22 17.5228 17.5228 22 12 22ZM12 20C16.4183 20 20 16.4183 20 12C20 7.58172 16.4183 4 12 4C7.58172 4 4 7.58172 4 12C4 16.4183 7.58172 20 12 20ZM11 7H13V9H11V7ZM11 11H13V17H11V11Z"
            />
          </svg>
        </div>
        <div
          class="flex flex-col gap-1"
        >
          <h3
            class="font-medium"
          >
            Alert Title
          </h3>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="alert p-4 rounded-lg border gap-3 text-sm bg-slate-50 border-slate-300 text-slate-600 [&h3]:text-slate-700"
      role="alert"
    >
      <div
        class="flex self-start pt-0.5"
      >
        <svg
          class="remixicon "
          fill="default"
          height="16"
          viewBox="0 0 24 24"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 22C6.47715 22 2 17.5228 2 12C2 6.47715 6.47715 2 12 2C17.5228 2 22 6.47715 22 12C22 17.5228 17.5228 22 12 22ZM12 20C16.4183 20 20 16.4183 20 12C20 7.58172 16.4183 4 12 4C7.58172 4 4 7.58172 4 12C4 16.4183 7.58172 20 12 20ZM11 7H13V9H11V7ZM11 11H13V17H11V11Z"
          />
        </svg>
      </div>
      <div
        class="flex flex-col gap-1"
      >
        <h3
          class="font-medium"
        >
          Alert Title
        </h3>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
